### DISCUSSION TITLE
Download error

### DISCUSSION THREAD

**Discussion by CurryAyam** [kolbasa/cordova-plugin-apkupdater#31] [Q&A] [answered]
I'm getting the following error while downloading the zip file

`message: "Download failed: https://storage.googleapis.com/davinci_images/android/ares2/app.zip (java.io.FileNotFoundException)"
stack: "de.kolbasa.apkupdater.exceptions.DownloadFailedException: Download failed: https://storage.googleapis.com/davinci_images/android/ares2/app.zip (java.io.FileNotFoundException)\n  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:255)\n  at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getInputStream(DelegatingHttpsURLConnection.java:211)\n  at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:30)\n  at de.kolbasa.apkupdater.downloader.FileDownloader.download(FileDownloader.java:62)\n  at de.kolbasa.apkupdater.update.UpdateManager.downloadFile(UpdateManager.java:76)\n  at de.kolbasa.apkupdater.update.UpdateManager.download(UpdateManager.java:124)\n  at de.kolbasa.apkupdater.ApkUpdater.download(ApkUpdater.java:110)\n  at de.kolbasa.apkupdater.ApkUpdater.lambda$execute$3$ApkUpdater(ApkUpdater.java:243)\n  at de.kolbasa.apkupdater.-$$Lambda$ApkUpdater$i2uPxQeilYT0voSmjrvq6lzNQe0.run(Unknown Source:6)\n  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)\n  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)\n  at java.lang.Thread.run(Thread.java:920)\n"
`

tried to use browser and the file exist

**Comment by kolbasa** [comment, marked as answer]
Thanks for the error description. I'll take a look at it now.

**Reply by CurryAyam** [reply]
I think my problem was using google cloud storage and AWS s3, sorry delete the previous comment.  :-) 👍

**Reply by kolbasa** [reply]
No problem.

The true error message was swallowed here: "Authentication required".  I need to fix the error handling here.

I will have a look at the Cloud SDKs in general. At least AWS, which is often requested.
